apps/hpc-ftsadmin/src/app/components/flow-form.tsx

```diff
@@ -0,0 +1,4094 @@
+    }
+  }, [comparingVersions]);
+
+  useEffect(() => {
+    const handleBeforeUnload = (event: BeforeUnloadEvent) => {
+      if (unsavedChange) {
+        const message =
+          'You have unsaved changes! Are you sure you want to leave?';
+        event.returnValue = message; // Standard for most browsers
+        return message; // For some older browsers
+      }
+    };
+
+    window.addEventListener('beforeunload', handleBeforeUnload);
+
+    return () => {
+      window.removeEventListener('beforeunload', handleBeforeUnload);
+    };
+  }, [unsavedChange]);
+
+  useEffect(() => {
+    const valuesObject: Record<string, any[]> = {
+      sourceOrganizations: initialValue.sourceOrganizations,
+      sourceLocations: initialValue.sourceLocations,
+      sourceUsageYears: initialValue.sourceUsageYears,
+      sourceProjects: initialValue.sourceProjects,
+      sourcePlans: initialValue.sourcePlans,
+      sourceGoverningEntities: initialValue.sourceGoverningEntities,
+      sourceGlobalClusters: initialValue.sourceGlobalClusters,
+      sourceEmergencies: initialValue.sourceEmergencies,
+      destinationOrganizations: initialValue.destinationOrganizations,
+      destinationLocations: initialValue.destinationLocations,
+      destinationUsageYears: initialValue.destinationUsageYears,
+      destinationProjects: initialValue.destinationProjects,
+      destinationPlans: initialValue.destinationPlans,
+      destinationGoverningEntities: initialValue.destinationGoverningEntities,
+      destinationGlobalClusters: initialValue.destinationGlobalClusters,
+      destinationEmergencies: initialValue.destinationEmergencies,
+    };
+    setObjects(valuesObject);
+  }, []);
+
+  unstable_usePrompt({
+    message: 'You have unsaved changes! Are you sure you want to leave?',
+    when: unsavedChange,
+  });
+  const dictExecutedForEachObject: Record<
+    string,
+    (
+      objectType: string,
+      flowObject: any,
+      setFieldValue: any,
+      values: any
+    ) => Promise<any>
+  > = {
+    sourcePlans: fetchPlanDetails,
+    destinationPlans: fetchPlanDetails,
+    destinationEmergencies: fetchEmergencyDetails,
+    sourceProjects: fetchProjectDetails,
+    destinationProjects: fetchProjectDetails,
+    sourceOrganizations: fetchOrganizationDetails,
+    sourceGlobalClusters: fetchAssociatedGoverningEntity,
+    destinationGlobalClusters: fetchAssociatedGoverningEntity,
+    sourceUsageYears: fetchKeywords,
+    destinationUsageYears: fetchKeywords,
+  };
+
+  const params: DeleteFlowParams = {
+    VersionID: currentVersionID ?? 0,
+    FlowID: currentFlowID ?? 0,
+  };
+
+  const deleteFlow = async () => {
+    if (linkCheck || isShowParentFlow || isShowChildFlow > 0) {
+      window.confirm(
+        'All linked flows must be unlinked before this flow can be deleted. Unlink flows and choose Delete flow again.'
+      );
+      return;
+    }
+
+    try {
+      const response = await env.model.flows.deleteFlow(params);
```

Comment:
try ... catch is not enough?

Reply:
Yeah! but I think using `catch()` will make your life easier to type the error instead of any.

Reply:
(When using `catch((err)=>...)` `err` will be `any` instead of `unknown` in the case of the `try ... catch`, this can make it easier to use our Error classes)